koku/masu/database/report_manifest_db_accessor.py

```diff
@@ -146,3 +146,39 @@ def get_last_seen_manifest_ids(self, bill_date):
             if not processed:
                 assembly_ids.append(manifest.assembly_id)
         return assembly_ids
+
+    def delete_cost_usage_reports_older_than(self, provider_type, billing_period_start_datetime):
+        """
+        Deletes Cost usage Reports older than billing_period_start_datetime.
+
+        Args:
+            provider_type   (String) the provider type to delete associated manifests
+            billing_period_start_datetime (datetime.datetime) delete all manifests older than this date, exclusive.
+        """
+        delete_count = CostUsageReportManifest.objects.filter(
+            provider__type=provider_type, billing_period_start_datetime__lt=billing_period_start_datetime
```

Comment:
Are we sure we want to delete all manifests the match the desired provider's type instead of the specific provider being removed?

Reply:
on second look. It looks like this is called when a provider_uuid is not supplied in `ExpiredDataRemover.remove()`